test/test_heuristic.cpp

```diff
@@ -81,6 +81,84 @@ TEST(Functionality, NodeCostCalculation) {
               tolerance);
 }
 
+TEST(Functionality, HeuristicBenchmark) {
+  const double tolerance = 1e-10;
+  /*
+      3
+     / \
+    4   2
+    |   |
+    0---1
+  */
+  Architecture      architecture{};
+  const CouplingMap cm = {{0, 1}, {1, 0}, {1, 2}, {2, 1}, {2, 3},
+                          {3, 2}, {3, 4}, {4, 3}, {4, 0}, {0, 4}};
+  architecture.loadCouplingMap(5, cm);
+
+  qc::QuantumComputation qc{5, 5};
+  qc.x(2, qc::Control{4});
+  qc.x(1, qc::Control{3});
+  qc.x(1, qc::Control{4});
+
+  qc.barrier({0, 1, 2, 3, 4});
+  for (size_t i = 0; i < 5; ++i) {
+    qc.measure(static_cast<qc::Qubit>(i), i);
+  }
+
+  const auto    mapper = std::make_unique<HeuristicMapper>(qc, architecture);
```

Comment:
Also just a side note: I am fairly sure that you could just allocate a plain `HeuristicMapper` object without the additional level of indirection.

Reply:
Well, I learned to always use `std::make_unique` if there is no good reason for not doing so (due to exception safety and such). But I have to admit that I am not an expert in such details of C++. So if you recommend using a direct instantiation here, I am going to change it.

Reply:
Nah. It should be fine to keep it. One of the main difference between the two variants is heap (with unique_ptr) vs stack (with plain constructor) allocation. IIRC, it can actually become a problem to create too large objects in the stack (our DD package for example). Overall, I think unique_ptr is probably the better solution here.